### ISSUE TITLE
Fix cut trees

### ISSUE THREAD

**Issue by sgdc3** [timvisee/DungeonMaze#42]
![2015-03-28_21 45 08](https://cloud.githubusercontent.com/assets/8779252/6882842/d1690372-d593-11e4-8299-6dcd65fbdb9e.png)

**Comment by timvisee** [comment]
Hello, 

This seems to be caused by chunks that aren't generated yet while the surface is being populated with trees. Once a tree is placed on the edge of the world the parts that are placed on non-generated chunks are being cut off. 

I currently don't know how to solve this issue. Possibly by storing the blocks that are generated 'outside' the world, an place them once the chunks are generated. But I don't know if this is a perfect solution. 

What do you think about this? 

Tim Visée

**Comment by sgdc3** [comment]
You can force the generation of a chunk where is a tree.

**Comment by timvisee** [comment]
That's true, but that could create an infinite chunk generation loop when you increase the tree density. Also, force-generating a chunk while another chunk is populating could break the generation process. 

So in short, this should be solved differently. 

Tim Visée
